### PR TITLE
Upgrade isort

### PR DESCRIPTION
**Pull request by st9007a**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix isort config. known_standard_library behavior have changed after 5.0.0.

## Modification

1. Upgrade `isort` to 5.10.1.
2. Replace known_standard_library with extra_standard_library in `isort` config for compatibility.
3. Remove `seed-isort-config` in linting CI. `seed-isort-config` is deprecated.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by YuanLiuuuuuu — APPROVED** [review]

**Review by YuanLiuuuuuu — CHANGES_REQUESTED** [review]
Since you change known_standard_library to extra_standard_library, isort may have different behaviors. Please run the pre-commit hooks to fix the lint.

**Comment by st9007a** [comment]
Thanks for remind.  All pre-commit hooks have been passed. The detail modification is described below.

I tracked the whole linting workflow in CI and found that the version of `isort` used in CI is 4.3.21. The latest version is 5.10.1 which is incompatible to 4.3.21. I think that upgrading `isort` to latest can avoid contributors confuse about unexpected formatting result if they didn't notice testing requirement.

To upgrade `isort` and make sure behavior is the same as previous version. I did two things:

1. Replace` known_standard_library` with `extra_standard_library` in `setup.cfg`. 

In [official document](https://github.com/PyCQA/isort/blob/0ccefec5a394a77a8a03e6c2f517b484765abcda/docs/upgrade_guides/5.0.0.md#known_standard_library), it suggests to use `extra_standard_library` in `isort>=5` to get the same behavior of `known_standard_library` in  `isort~=4`.


2. Remove `seed-isort-config` hook in linting CI.

GitHub [README](https://github.com/asottile-archive/seed-isort-config) of `seed-isort-config` has mentioned that it is no longer needed as of `isort>=5`.

**Comment by st9007a** [comment]
To align modification and this pull request, the title and modification section was edited.

File: mmselfsup/datasets/builder.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def build_datasource(cfg, default_args=None):
 
 
 def build_dataset(cfg, default_args=None):
-    from .dataset_wrappers import (ConcatDataset, RepeatDataset)
+    from .dataset_wrappers import ConcatDataset, RepeatDataset
```
Comment (YuanLiuuuuuu) [inline mmselfsup/datasets/builder.py:35]: Currently, we have not created unit test for `build_dataset`, so you would not pass the Codecov. Could you please help us to write the unit test func in the `test` folder ?

File: tests/test_data/test_datasets/test_builder.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import pytest
```
Comment (YuanLiuuuuuu) [inline tests/test_data/test_datasets/test_builder.py:3]: Thank you for providing the  unit test code for build_dataset. There are still some problems when runing unit test and we highly suggest you to run `pytest` in your local machine before pushing your updated code to this PR.

Comment (st9007a) [reply tests/test_data/test_datasets/test_builder.py:3]: Yes, I am trying. The latest commit works on my local machine. Please check it.

Comment (YuanLiuuuuuu) [reply tests/test_data/test_datasets/test_builder.py:3]: All checks have passed. Thank you ~

**Pull request merged**